### PR TITLE
Js store

### PR DESCRIPTION
**Pull request by aricart**

### CODE REVIEW

File: providers/jetstream/jsprovider.go [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+		p.disconnect()
+		return err
+	}
+	p.js = js
+	_, err = p.js.AccountInfo(context.Background())
+	if err != nil {
+		p.disconnect()
+		return err
+	}
+	p.kv, err = p.js.KeyValue(context.Background(), p.bucket)
```
Comment (ripienaar) [inline providers/jetstream/jsprovider.go:59]: should maye accept a context on `NewJetstreamProvider()` and get rid of all the `context.Background()`

File: providers/jetstream/jsprovider.go [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+
+func (p *Provider) disconnect() {
+	p.nc.Close()
+}
+
+// child entities are stored under <pk>.<childPublicKey>
+func (p *Provider) getChildren(prefix string) (map[string][]byte, error) {
+	entries, err := p.kv.History(context.Background(), fmt.Sprintf("%s.*", prefix))
+	if err != nil {
+		if err == jetstream.ErrKeyNotFound {
```
Comment (ripienaar) [inline providers/jetstream/jsprovider.go:85]: should now use `errors.Is(err, jetstream.ErrKeyNotFound)`

File: providers/jetstream/jsprovider.go [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+		if err := p.loadUsers(a); err != nil {
+			return err
+		}
+		od.AccountDatas = append(od.AccountDatas, a)
+	}
+	return nil
+}
+
+func (p *Provider) loadUsers(ad *ab.AccountData) error {
+	// users stored under <accountPublicKey>.<userPublicKey>
```
Comment (ripienaar) [inline providers/jetstream/jsprovider.go:182]: be good to get a little README or something that pulls this design out into a single place

File: providers/jetstream/jsprovider.go [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+			return err
+		}
+
+		ad.UserDatas = append(ad.UserDatas, u)
+	}
+	return nil
+}
+
+func (p *Provider) getKey(pk string) (*ab.Key, error) {
+	e, err := p.kv.Get(context.Background(), fmt.Sprintf("keys.%s", pk))
```
Comment (ripienaar) [inline providers/jetstream/jsprovider.go:212]: Do you think we should encrypt these keys before putting them in kv? I wish we had the codecs we planned for KV sigh so this whole store could just be encrypted

Comment (aricart) [reply providers/jetstream/jsprovider.go:212]: I think if they provide a key to the provider that it should do that.

**Pull request merged**